backend/src/main/java/services/CertificateService.java

```diff
@@ -1,4 +1,98 @@
 package services;
 
+import data.entities.CertificateEntity;
+import data.entities.CertificateType;
+import data.entities.CommentEntity;
+import data.entities.UserEntity;
+import data.repositories.CertificateRepository;
+import data.repositories.UserRepository;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.NotFoundException;
+import web.dtos.CertificateDto;
+import web.mappers.CertificateMapper;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@Transactional
 public class CertificateService {
-}
+
+    @Inject
+    CertificateMapper certificateMapper;
+    @Inject
+    CertificateRepository certificateRepository;
+    @Inject
+    UserRepository userRepository;
+
+    public List<CertificateDto> getCertificates(){
+        List<CertificateEntity> certificateEntities = certificateRepository.listAll();
+        return certificateEntities.stream()
+                .map(certificateMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    public CertificateDto createCertificate(CertificateDto certificateDto) {
+        CertificateEntity certificateEntity = certificateMapper.toEntity(certificateDto);
+        certificateRepository.persist(certificateEntity);
+        return certificateMapper.toDto(certificateEntity);
+    }
+
+    public CertificateDto updateCertificateDto(Long id, CertificateDto certificateDto) {
+        CertificateEntity existingCertificateEntity = certificateRepository.findById(id);
+
+        if (existingCertificateEntity == null) {
+            throw new NotFoundException("Certificate with id " + id + " not found");//for now let's use this notfound by the jakarta.ws.rs.NotFoundException;
```

Comment:
Not just for now, NotFoundException is correct choice here anyway

Reply:
> Not just for now, NotFoundException is correct choice here anyway

I was thinking i am going to be using my class that implements it.